### PR TITLE
feat: support config.yml for global setting

### PR DESCRIPTION
**Pull request by k1LoW**
This pull request introduces support for global configuration files in the `deck` tool, enabling users to define default settings and conditional actions for presentations at an organization- or project-wide level. The changes include updates to the documentation, new configuration handling logic, and corresponding tests. Below are the key updates:

### Documentation Enhancements
* Updated `README.md` to include details about global configuration files, their precedence, and an example configuration file. This explains how users can set global defaults for `breaks` and `defaults` fields and override them using frontmatter. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L135-R181) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R199-R208)

### Configuration Management
* Introduced a new `config` package with a `Config` struct and a `Load` function to handle loading configuration files from predefined paths. This includes support for profiles and YAML parsing.
* Added the ability to apply configuration settings to markdown files using the `ApplyConfig` method in the `md` package. This method merges global configuration with frontmatter settings.

### Code Changes in Command Logic
* Modified `cmd/apply.go` to load the configuration file and apply its settings before processing markdown files. This ensures that global defaults are respected during execution. [[1]](diffhunk://#diff-a08f2011636c2b27ac4df1809b3bf95ee8aa9a6de2050807d33ddfb65fb69f6fL82-R91) [[2]](diffhunk://#diff-a08f2011636c2b27ac4df1809b3bf95ee8aa9a6de2050807d33ddfb65fb69f6fL150-R154) [[3]](diffhunk://#diff-a08f2011636c2b27ac4df1809b3bf95ee8aa9a6de2050807d33ddfb65fb69f6fL258-R262) [[4]](diffhunk://#diff-a08f2011636c2b27ac4df1809b3bf95ee8aa9a6de2050807d33ddfb65fb69f6fR327)

### Frontmatter Updates
* Changed the `Breaks` field in the `Frontmatter` struct to a pointer (`*bool`) to differentiate between unset and explicitly set values. This allows better handling of default values. [[1]](diffhunk://#diff-f10e982d75ada71772b6eea3f470a79ccdb670e48b51e202deb638bf61b80517L54-R55) [[2]](diffhunk://#diff-f10e982d75ada71772b6eea3f470a79ccdb670e48b51e202deb638bf61b80517L144-R146)

### Testing Improvements
* Added comprehensive unit tests in `md_test.go` to verify the behavior of the `ApplyConfig` method, ensuring correct merging of configuration and frontmatter settings. This includes edge cases like appending default conditions and preserving existing settings. [[1]](diffhunk://#diff-a875d3534d9727c49cb3453f84caac0054354cf1d674fd78fd638afc7593ecf8R73-R204) [[2]](diffhunk://#diff-a875d3534d9727c49cb3453f84caac0054354cf1d674fd78fd638afc7593ecf8R284-R288)

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1,15 +1,60 @@
...
+			if b, err := os.ReadFile(configPath); err == nil {
+				if err := yaml.Unmarshal(b, cfg); err != nil {
+					return nil, fmt.Errorf("failed to unmarshal config: %w", err)
+				}
+				return cfg, nil
+			}
+		}
+	}
+	// If no config file is found, return an empty config
+	return cfg, nil
```
[the remark this answers is not in the document]
Comment (k1LoW) [reply config/config.go:55]: I'll keep it as is.
